viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/process/strategy/TerminatedTwoTrackSectionSynchronizer.scala

```diff
@@ -0,0 +1,458 @@
+
+    // Compare original addresses here because we are modifying the starting situation
+    val continuousAfterTerminatedLeft   = projectLinks.find(pl => pl.track == Track.LeftSide && pl.originalAddrMRange.continuesFrom(lastTerminatedOnLeftSideSection.originalAddrMRange))
+    val continuousAfterTerminatedRight  = projectLinks.find(pl => pl.track == Track.RightSide && pl.originalAddrMRange.continuesFrom(lastTerminatedOnRightSideSection.originalAddrMRange))
+
+    // Adjust terminated
+    val processedLinks: Seq[ProjectLink] = {
+      if (areTracksCloseEnoughOnEndAddrM(lastTerminatedOnLeftSideSection, lastTerminatedOnRightSideSection)) {
+        // Adjust the last terminated links on each track to match
+        val (adjustedTerminatedLeft, adjustedTerminatedRight, averagedAddrMRange) = adjustTerminatedToMatch(lastTerminatedOnLeftSideSection, lastTerminatedOnRightSideSection)
+
+        if (continuousAfterTerminatedLeft.isDefined && continuousAfterTerminatedRight.isDefined) {
+          // Adjust both links after the terminated section
+          val adjustedLeftAddrMRange = AddrMRange(averagedAddrMRange.end, continuousAfterTerminatedLeft.get.originalAddrMRange.end)
+          val adjLeftContinuousAfterTerminated = continuousAfterTerminatedLeft.get.copy(
+            originalAddrMRange = adjustedLeftAddrMRange
+          )
+
+          val adjustedRightAddrMRange = AddrMRange(averagedAddrMRange.end, continuousAfterTerminatedRight.get.originalAddrMRange.end)
+          val adjRightContinuousAfterTerminated = continuousAfterTerminatedRight.get.copy(
+            originalAddrMRange = adjustedRightAddrMRange
+          )
+          // Update the project links list with the adjusted terminated links and the adjusted continuous-after-termination links
+          updateProjectLinksList(Seq(adjustedTerminatedLeft, adjustedTerminatedRight, adjLeftContinuousAfterTerminated, adjRightContinuousAfterTerminated), projectLinks)
+        } else {
+          // Just update the terminated links
+          updateProjectLinksList(Seq(adjustedTerminatedLeft, adjustedTerminatedRight), projectLinks)
+        }
+
+      } else {
+        projectLinks
+      }
+    }
+    processedLinks
+  }
+
+  private def roadPartMiddleTerminated(projectLinks: Seq[ProjectLink]): Seq[ProjectLink] = {
+
+    val terminatedLinks = projectLinks.filter(_.status == RoadAddressChangeType.Termination)
+
+    val (leftTerminatedSections, rightTerminatedSections) = terminatedLinksToContinuousTwoTrackSections(terminatedLinks)
+
+    val processedLinks = if (leftTerminatedSections.nonEmpty && rightTerminatedSections.nonEmpty) { // Check if road part start is terminated on both tracks
+      handleTwoTrackMiddleTermination(leftTerminatedSections, rightTerminatedSections, projectLinks)
+    } else
+      projectLinks
+
+    processedLinks
+  }
+
+  /**
+   * Adjusts two track terminated sections to match if the termination creates a minor discontinuity.
```

Comment:
"--- if the termination had created a minor discontinuity before the termination."

Reply:
"--- if the link or links preceding the termination have a minor discontinuity "